### PR TITLE
Allow setting the single attribute merge strategy

### PR DESCRIPTION
**Pull request by jcazevedo**
This PR adds the `setMergeAttribute` method to the User Profile Storage client, in order to allow specifying how specific attributes should be merged. Attributes can be merged when using the `update` method with the `JoinMergingStrategy`, or when two users are linked.

### CODE REVIEW

File: sdk/src/main/scala/eu/shiftforward/adstax/ups/api/SingleAttributeMergingStrategy.scala [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+ * [[NumericAttrValue]]s with value equal to `0` are converted to [[BooleanAttrValue]]s with value `false`.
+ * [[StringAttrValue]]s with value `"true"` are converted to [[BooleanAttrValue]]s with value `true`.
+ * [[StringAttrValue]]s with value `"false"` are converted to [[BooleanAttrValue]]s with value `false`.
+ * [[StringAttrValue]]s whose value can be parsed to a Double follow the rules of the [[NumericAttrValue]] conversion
+ * enumerated above.
+ */
+case object Or extends SingleAttributeMergingStrategy
+
+object SingleAttributeMergingStrategy {
+  implicit object SingleAttributeMergingStrategyJsonFormat extends JsonFormat[SingleAttributeMergingStrategy] {
```
Comment (ruippeixotog) [inline sdk/src/main/scala/eu/shiftforward/adstax/ups/api/SingleAttributeMergingStrategy.scala]: I have to create an `EnumCoproductHint` in pureconfig for those cases :(

File: sdk/src/main/scala/eu/shiftforward/adstax/ups/api/UserProfileStorageClient.scala [hunk 2]
```diff
@@ -80,6 +80,16 @@ trait UserProfileStorageClient {
   def link(userId1: UserId, userId2: UserId, source: Option[String] = None): Future[Unit]
 
   /**
+   * Sets the merge strategy for a single attribute. If an attribute doesn't have a merging strategy defined, it is
```
Comment (ruippeixotog) [inline sdk/src/main/scala/eu/shiftforward/adstax/ups/api/UserProfileStorageClient.scala]: Can you specify that this applies only to base attributes ("...merge strategy for a single base attribute.")? Even if it doesn't make sense to do that with computed attributes, it's better if this is kept clear in the documentation.

File: sdk/src/main/scala/eu/shiftforward/adstax/ups/api/AttributesMergingStrategy.scala [hunk 3]
```diff
@@ -14,27 +14,25 @@ sealed trait AttributeMergingStrategy
  * Example:
  * [(a, 1), (b, 1)] * [(a, 3), (c, 1)] = [(a, 3), (b, 1), (c, 1)]
  */
-case object JoinMergingStrategy extends AttributeMergingStrategy
+case object JoinMergingStrategy extends AttributesMergingStrategy
```
Comment (ruippeixotog) [inline sdk/src/main/scala/eu/shiftforward/adstax/ups/api/AttributesMergingStrategy.scala]: This is the strategy to apply in order for the UPS to deal with single attribute merging strategies. But this is called `JoinMergingStrategy`, while the method to set single attribute strategies is called `setMergeStrategy`. There is yet another `Merge` strategy, which is a single attribute strategy and only works with maps.

Can we make this less confusing by calling this `MergeMergingStrategy` (ok, confusing in its own way :P) and call `MapMerge` to its single attribute friend?

File: build.sbt [hunk 4]
```diff
@@ -55,7 +55,7 @@ enablePlugins(CrossPerProjectPlugin)
 
 lazy val commonSettings = Seq(
   organization := "eu.shiftforward",
-  version := "0.5.4",
+  version := "0.6.0-SNAPSHOT",
```
Comment (ruippeixotog) [inline build.sbt]: Can you do a commit setting the release version?

File: sdk/src/main/scala/eu/shiftforward/adstax/ups/api/UserProfileStorageRequest.scala [hunk 5]
```diff
@@ -95,6 +95,18 @@ object LinkUsers {
 }
 
 /**
+ * A request to set the merge strategy of a given attribute.
```
Comment (ruippeixotog) [inline sdk/src/main/scala/eu/shiftforward/adstax/ups/api/UserProfileStorageRequest.scala]: The same thing applies here regarding the mention to base attributes.

File: sdk/src/main/scala/eu/shiftforward/adstax/ups/api/UserProfileStorageClient.scala (hunk 2 above)
Comment (jcazevedo) [inline sdk/src/main/scala/eu/shiftforward/adstax/ups/api/UserProfileStorageClient.scala]: Improved documentation in 050c7d8.

File: sdk/src/main/scala/eu/shiftforward/adstax/ups/api/UserProfileStorageRequest.scala (hunk 5 above)
Comment (jcazevedo) [inline sdk/src/main/scala/eu/shiftforward/adstax/ups/api/UserProfileStorageRequest.scala]: Improved documentation in 050c7d8.

File: sdk/src/main/scala/eu/shiftforward/adstax/ups/api/AttributesMergingStrategy.scala (hunk 3 above)
Comment (jcazevedo) [inline sdk/src/main/scala/eu/shiftforward/adstax/ups/api/AttributesMergingStrategy.scala]: Changed names in 050c7d8.

File: build.sbt (hunk 4 above)
Comment (jcazevedo) [inline build.sbt]: I'll do it once this gets merged in, so that I can use the `release` command.

**Pull request merged**